code/code/task/task_picklock.cc

```diff
@@ -1,144 +1,122 @@
-//////////////////////////////////////////////////////////////////////////
-//
-//      SneezyMUD 4.0 - All rights reserved, SneezyMUD Coding Team
-//      "task.cc" - All functions related to tasks that keep mobs/PCs busy
-//
-//////////////////////////////////////////////////////////////////////////
-
 #include "extern.h"
 #include "comm.h"
 #include "room.h"
 #include "obj_tool.h"
 #include "being.h"
 
-void TThing::pickPulse(TBeing *ch)
-{
-  ch->sendTo("Hey, where'd your lockpick go?!?\n\r");
-  ch->stopTask();
-}
-
-void TTool::pickPulse(TBeing *ch)
+void pick_pulse(TBeing *ch, TThing *pick)
 {
-  int skill,difficulties;
-  TRoom *temp_rp;
-  roomDirData *exitp, *back;
-
-  if (getToolType() != TOOL_LOCKPICK) {
+  TTool *tool = dynamic_cast<TTool *>(pick);
+  if (!tool || tool->getToolType() != TOOL_LOCKPICK) {
     ch->sendTo("Hey, where'd your lockpick go?!?\n\r");
     ch->stopTask();
     return;
   }
-  if (!(exitp = ch->roomp->dir_option[ch->task->status])) {
+
+  roomDirData *exit;
+  if (!(exit = ch->roomp->dir_option[ch->task->status])) {
     ch->sendTo("Errr, what happened to the exit???\n\r");
     ch->stopTask();
     return;
   }
-  if (!IS_SET(exitp->condition, EXIT_LOCKED)) {
+
+  if (!IS_SET(exit->condition, EXIT_LOCKED)) {
     ch->sendTo("Some nice person seems to have unlocked it for you.\n\r");
     ch->stopTask();
     return;
   }
-  skill = 2 * ch->getSkillValue(SKILL_PICK_LOCK);
 
-  // longer they try, better chance
   ch->task->timeLeft++;
-  skill += 3 * ch->task->timeLeft;
+  int skill = 2 * ch->getSkillValue(SKILL_PICK_LOCK)
+      + 3 * ch->task->timeLeft // longer they try, better chance
+      + ch->plotStat(STAT_CURRENT, STAT_FOC, 3, 18, 13)
+      + ::number(-15,15);
 
-  skill += ch->plotStat(STAT_CURRENT, STAT_FOC, 3, 18, 13);
-  skill += ::number(-15,15);
+  if (!tool->addToToolUses(-1)) {
+    act("Your $o snap in half!   Oops!", true, ch, tool, nullptr, TO_CHAR);
+    act("$n's $o snaps in half.", true, ch, tool, nullptr, TO_ROOM);
```

Comment:
Probably should be singular?

Reply:
oh! there's format codes for this kind of thing, let me use them... except that lockpicks are plural in the strings and singular in the code - an unfortunate mismatch. I will add code to handle that. the string fixup is added first.

Reply:
added a TTool overload for isPluralItem, that checks if fname(name) ends in 's', which wraps this up. then if someone makes a singular lockpick we're covered.

Reply:
oh! I'd missed that there's a skeleton key item, with fname "key", and it'll show as singular! perfect!